service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java

```diff
@@ -347,6 +358,66 @@ public void run() {
         }
     }
 
+    // TODO: move this function to TempoService and TempoServiceImpl? (except for the publish part)
+    // TODO: break this up into smaller functions?
+    private void publishTempoSamplesToCBioPortal(Set<String> tumorPrimaryIds) {
```

Comment:
What do you think about (a) breaking this function into smaller functions, and (b) moving the definitions of this block of logic to either the existing `TempoService` or a new `TempoMessagePublishingService`?

For (a), the smaller functions could be:
```java
private Set<TempoSamples> getValidTempoSamples(Set<String> tumorPrimaryIds) throws Exception;
private TempoSample buildTempoSample(String primaryId, String cmoSampleName, String accessLevel, String custodianInformation) throws Exception;
private TempoSampleUpdateMessage buildTempoSampleUpdateMessage(Set<TempoSamples> validTempoSamples) throws Exception;
```

The reasoning for (b) is that this file's name indicates that it's for message handling/receiving, while this logic is for message publishing.